backend-api/template.yaml

```diff
@@ -243,6 +243,95 @@ Resources:
           Parameters:
             Location: ./openApiSpecs/async-private-spec.yaml
 
+  PrivateApiDomainName:
+    Type: AWS::ApiGateway::DomainName
+    Properties:
+      DomainName: !Sub
+        - private-${AWS::StackName}.${DNS_RECORD}
```

Comment:
@darrendunford - did you have any opinions on naming? 

This would come through as private-mob-async-backend.${DNS_RECORD}. I was potentially wondering about this being 'oauth-mob-async-backend.${DNS_RECORD}

Reply:
since we're using a public hosted zone for this the DNS name is visible to the world. I guess `oauth-...` would be better as the word `private` may attract underwanted attention from those who seek to gain access to something private ???